<?xml version="1.0" encoding="UTF-8"?>

<launch>
  <rosparam file="$(find carla_circle)/config/control_settings.yaml" command="load" />

  <arg name="time_step" default="0.3" />
  <arg name="plan_steps" default="7" />
  <arg name="ref_speed" default="5.0" />

  <!-- share one bridge  -->
	<!-- <include file="$(find carla_ros_bridge)/launch/carla_ros_bridge.launch">
    <arg name="ego_vehicle_name" value="['stanford_ego', 'stanford_hero']" />
	</include> -->


  <!-- launch main components  -->
  <group ns="stanford_hero" >
    <node name="controller" pkg="carla_circle" type="fancy_controller.py" output="screen">
      <param name="rolename" value="stanford_hero" />
      <param name="time_step" value="$(arg time_step)" />
      <param name="ctrl_freq" value="100" />
      <param name="plan_steps" value="$(eval arg('plan_steps') + 1)" />
      <param name="ref_speed" value="$(arg ref_speed)" />
    </node>

    <node name="supervisor_node" pkg="carla_circle" type="supervisor.py" output="screen">
      <param name="rolename" value="stanford_hero" />
    </node>
    <node name="viz_node" pkg="carla_circle" type="visualization_node.py" output="screen" />
  </group>

  <group ns="stanford_ego" >
    <node name="controller" pkg="carla_circle" type="fancy_controller.py" output="screen">
      <param name="rolename" value="stanford_ego" />
      <param name="time_step" value="$(arg time_step)" />
      <param name="ctrl_freq" value="100" />
      <param name="plan_steps" value="$(eval arg('plan_steps') + 1)" />
      <param name="ref_speed" value="$(arg ref_speed)" />
    </node>

    <node name="supervisor_node" pkg="carla_circle" type="supervisor.py" output="screen">
      <param name="rolename" value="stanford_ego" />
    </node>
    <node name="viz_node" pkg="carla_circle" type="visualization_node.py" output="screen" />
  </group>


	<!-- rviz window -->
	<node pkg="rviz" type="rviz" name="rviz" args="-d $(find carla_ros_bridge)/config/carla_default_rviz.cfg.rviz"/>

  <!-- for debugging -->
  <node pkg="leq_game" type="debug_viz.py" name="speed_debug" output="screen" />
</launch>
